File: sample/src/main/java/com/agilie/sample/DrawingActivity.java
package com.agilie.sample;

import android.content.Context;
import android.graphics.Typeface;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.RelativeLayout;

import com.agilie.RotatableAutofitEditText;

import java.util.ArrayList;

public class DrawingActivity extends AppCompatActivity {

    private Button addTextButton;
    private Button addInsomniaRegularTextButton;
    private Button addDroidRobotTextButton;
    private RelativeLayout container;
    private ArrayList<RotatableAutofitEditText> editTexts = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_drawing);

        container = (RelativeLayout) findViewById(R.id.container);
        addTextButton = (Button) findViewById(R.id.addTextButton);
        addInsomniaRegularTextButton = (Button) findViewById(R.id.addInsomniaRegularTextButton);
        addDroidRobotTextButton = (Button) findViewById(R.id.addDroidRobotTextButton);


        addTextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               addTextButton();
            }
        });

        addInsomniaRegularTextButton.setTypeface(Typeface.createFromAsset(getAssets(), "fonts/android_insomnia_regular.ttf"));
        addInsomniaRegularTextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                addTextButton(Typeface.createFromAsset(getAssets(), "fonts/android_insomnia_regular.ttf"));
            }
        });

        addDroidRobotTextButton.setTypeface(Typeface.createFromAsset(getAssets(), "fonts/doridrobot.ttf"));
        addDroidRobotTextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                addTextButton(Typeface.createFromAsset(getAssets(), "fonts/doridrobot.ttf"));
            }
        });

    }

    private void addTextButton(Typeface typeface) {
        RotatableAutofitEditText newEditText = (RotatableAutofitEditText) LayoutInflater.from(DrawingActivity.this).inflate(R.layout.view_autoresize_edittext, container, false);
        container.addView(newEditText);
        newEditText.requestLayout();
        newEditText.setTypeface(typeface);
        editTexts.add(newEditText);
        showSoftKeyboard(newEditText);
    }

    private void addTextButton() {
        RotatableAutofitEditText newEditText = (RotatableAutofitEditText) LayoutInflater.from(DrawingActivity.this).inflate(R.layout.view_autoresize_edittext, container, false);
        container.addView(newEditText);
        newEditText.requestLayout();
        editTexts.add(newEditText);
        showSoftKeyboard(newEditText);
    }

    private void showSoftKeyboard(View view) {
        if (view.requestFocus()) {
            InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT);
        }
    }

}

File: sample/src/main/java/com/agilie/sample/DrawingActivity.kt
package com.agilie.sample

import android.content.Context
import android.graphics.Typeface
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.RelativeLayout


import com.agilie.RotatableAutofitEditText
import kotlinx.android.synthetic.main.activity_drawing.*

import java.util.ArrayList

class DrawingActivity : AppCompatActivity() {

    private val editTexts = ArrayList<RotatableAutofitEditText>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_drawing)
        addTextButton!!.setOnClickListener { addTextButton() }

        addInsomniaRegularTextButton!!.typeface = Typeface.createFromAsset(assets, "fonts/android_insomnia_regular.ttf")
        addInsomniaRegularTextButton!!.setOnClickListener { addTextButton(Typeface.createFromAsset(assets, "fonts/android_insomnia_regular.ttf")) }

        addDroidRobotTextButton!!.typeface = Typeface.createFromAsset(assets, "fonts/doridrobot.ttf")
        addDroidRobotTextButton!!.setOnClickListener { addTextButton(Typeface.createFromAsset(assets, "fonts/doridrobot.ttf")) }

    }

    private fun addTextButton(typeface: Typeface) {
        val newEditText = LayoutInflater.from(this@DrawingActivity).inflate(R.layout.view_autoresize_edittext, container, false) as RotatableAutofitEditText
        container!!.addView(newEditText)
        newEditText.requestLayout()
        newEditText.typeface = typeface
        editTexts.add(newEditText)
        showSoftKeyboard(newEditText)
    }

    private fun addTextButton() {
        val newEditText = LayoutInflater.from(this@DrawingActivity).inflate(R.layout.view_autoresize_edittext, container, false) as RotatableAutofitEditText
        container!!.addView(newEditText)
        newEditText.requestLayout()
        editTexts.add(newEditText)
        showSoftKeyboard(newEditText)
    }

    private fun showSoftKeyboard(view: View) {
        if (view.requestFocus()) {
            val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.showSoftInput(view, InputMethodManager.SHOW_IMPLICIT)
        }
    }

}
